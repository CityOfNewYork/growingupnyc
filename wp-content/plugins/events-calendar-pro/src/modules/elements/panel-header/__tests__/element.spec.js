import React from 'react';
import renderer from 'react-test-renderer';
import PanelHeader from '../element';

describe( 'PanelHeader', () => {
	test( 'should render component with count', () => {
		const component = renderer.create(
			<PanelHeader
				count={ 2 }
				isExpanded={ false }
				onClick={ jest.fn() }
			>
				<span>Test children</span>
			</PanelHeader>
			);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render component without count', () => {
		const component = renderer.create(
			<PanelHeader
				count={ 2 }
				isExpanded={ true }
				onClick={ jest.fn() }
			>
				<span>Test children</span>
			</PanelHeader>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render component without count when count is 0', () => {
		const component = renderer.create(
			<PanelHeader
				count={ 0 }
				isExpanded={ false }
				onClick={ jest.fn() }
			>
				<span>Test children</span>
			</PanelHeader>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
