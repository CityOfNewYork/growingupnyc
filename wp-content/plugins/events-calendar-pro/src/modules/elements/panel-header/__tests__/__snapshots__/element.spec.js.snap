// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelHeader should render component with count 1`] = `
<header
  className="tribe-editor__events-pro__panel-header"
>
  <button
    className="tribe-editor__events-pro__panel-header-button"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="tribe-editor__events-pro__panel-header-button-text"
    >
      <span
        className="tribe-editor__events-pro__panel-header-button-title"
      >
        <span>
          Test children
        </span>
      </span>
      <span
        className="tribe-editor__events-pro__panel-header-button-count"
      >
        (2)
      </span>
    </span>
    <<svg>icon mock</svg> />
  </button>
</header>
`;

exports[`PanelHeader should render component without count 1`] = `
<header
  className="tribe-editor__events-pro__panel-header tribe-editor__events-pro__panel-header--expanded"
>
  <button
    className="tribe-editor__events-pro__panel-header-button"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="tribe-editor__events-pro__panel-header-button-text"
    >
      <span
        className="tribe-editor__events-pro__panel-header-button-title"
      >
        <span>
          Test children
        </span>
      </span>
    </span>
    <<svg>icon mock</svg> />
  </button>
</header>
`;

exports[`PanelHeader should render component without count when count is 0 1`] = `
<header
  className="tribe-editor__events-pro__panel-header"
>
  <button
    className="tribe-editor__events-pro__panel-header-button"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="tribe-editor__events-pro__panel-header-button-text"
    >
      <span
        className="tribe-editor__events-pro__panel-header-button-title"
      >
        <span>
          Test children
        </span>
      </span>
    </span>
    <<svg>icon mock</svg> />
  </button>
</header>
`;
